File: CommonBase/src/main/java/com/common/base/mvp/MvpActivity.java
package com.common.base.mvp;

import android.os.Bundle;
import com.common.base.BaseActivity;
import com.common.base.BasePresenter;
import com.common.base.ability.IPresenter;

/**
 * MVP模式Activity
 *
 * @author LiuFeng
 * @data 2021/9/22 16:19
 */
public abstract class MvpActivity<P extends BasePresenter> extends BaseActivity implements IPresenter<P> {

    protected P mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        this.mPresenter = this.createPresenter();
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (this.mPresenter != null) {
            this.mPresenter.onDestroy();
        }
    }
}

File: CommonBase/src/main/java/com/common/base/mvp/MvpActivity.kt
package com.common.base.mvp

import android.os.Bundle
import com.common.base.BaseActivity
import com.common.base.ability.IBaseView

/**
 * MVP模式Activity
 *
 * @author LiuFeng
 * @data 2021/9/22 16:19
 */
abstract class MvpActivity<P : BasePresenter<out IBaseView>> : BaseActivity(), IPresenter<P> {
    protected lateinit var mPresenter: P

    override fun onCreate(savedInstanceState: Bundle?) {
        mPresenter = createPresenter()
        super.onCreate(savedInstanceState)
    }

    override fun onDestroy() {
        super.onDestroy()
        mPresenter.onDestroy()
    }
}